.eslintrc.js

```diff
@@ -1,21 +1,40 @@
-"use strict";
+/**
+ * @file    .eslintrc.js - Configures ESLint for the plugin repo.
+ *
+ * @author  Joash Mathew <joash@distributive.network>
+ * @author  Bryan Hoang <bryan@distributive.network>
+ * @date    June 2023, Aug. 2023
+ */
+'use strict';
 
+/** @type {import('eslint').Linter.Config} */
 module.exports = {
   root: true,
   extends: [
-    "eslint:recommended",
-    "plugin:eslint-plugin/recommended",
-    "plugin:node/recommended",
+    'eslint:recommended',
+    'plugin:n/recommended',
+    'plugin:eslint-plugin/tests',
+    'plugin:eslint-plugin/rules-recommended',
+    '@distributive',
+    'plugin:@distributive/recommended',
   ],
   env: {
     node: true,
-    es6: true,
-    browser: true
   },
-  overrides: [
-    {
-      files: ["tests/**/*.js"],
-      env: { mocha: true },
+  overrides: [ {
+    files: ['**/*.mjs'],
```

Comment:
Yep! Since the [`"types"`](https://nodejs.org/api/packages.html#type) field of the `package.json` file is omitted, all `.js` files by default are CommonJS modules when loaded by Node. But Vitest's config file format has to be an ECMAScript Module, so the `.mjs` extension tells Node.js to load it as an ES module.

ESLint needs to be configured to understand that behaviour, unfortunately. It'd be nice if it understood that behaviour by default.